### PR TITLE
Initial theme mode

### PR DESCRIPTION
**Pull request by mohammedhemaid**
The video of the feature:
https://photos.app.goo.gl/UXhRZQybrh5bKHVz5

For the fragments recreation, we can use the old instance but that requires more work:
1- override onSavedInstanceState and save the currentViewId for the selected fragment.
2- on recreation  in OnCreate we will assign the homepageFragment and settingsFragment in the 
 HomepageFragmentController with the old instance from the FragmentManager.
3- In the BottomNavigationView we will need to select the selected fragment.

### CODE REVIEW

File: app/src/main/java/com/randomappsinc/studentpicker/settings/SettingsFragment.java [hunk 1]
```diff
@@ -94,6 +96,15 @@ public void onItemClick(int position) {
                 preferencesManager.setShakeEnabled(!currentState);
                 return;
             case 2:
+                int darkModePosition = preferencesManager.isOnFreeVersion() ? 2 : 1;
+                View secondCell = settingsOptions.getChildAt(darkModePosition);
+                Switch DarkModeToggle = secondCell.findViewById(R.id.shake_toggle);
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/settings/SettingsFragment.java]:
```suggestion
                Switch darkModeToggle = secondCell.findViewById(R.id.shake_toggle);
```

File: app/src/main/res/values/settings_strings.xml [hunk 2]
```diff
@@ -3,13 +3,15 @@
     <string name="buy_premium">Buy premium</string>
     <string name="buy_button_text">Buy ($0.99)</string>
     <string name="shake_enabled">Shake to choose</string>
+    <string name="dark_Mode">Dark Mode</string>
```
Comment (Gear61) [inline app/src/main/res/values/settings_strings.xml]:
```suggestion
    <string name="dark_mode">Dark mode</string>
```

**Review by Gear61 — CHANGES_REQUESTED** [review]
This looks great! The code makes a lot of sense to me, and it's quite clean overall.

Anyways, after watching the video, it's a pretty bad experience when you toggle the dark mode switch and you get taken back to the homepage. Does onSavedInstanceState() get called when the app restarts? We could theoretically call getActivity() and call some sort of setter on it so that when it restarts (savedInstanceState Bundle is non-null in onCreate()), it knows to tab to the settings fragment. Is this feasible? Let's try to make this user experience work, even if we have to write some messy code. We can always wrap the code to make it not as messy, haha.

**Comment by mohammedhemaid** [comment]
The video of the update:
https://photos.app.goo.gl/BjGTYgdb5Vqq2pwf9

**Comment by mohammedhemaid** [comment]
@Gear61 
I was worried about the messy code hahaha,  now I used the old instance and get the id of the latest selected view, it now working but the code still needs some cleaning.

File: app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java [hunk 3]
```diff
@@ -49,7 +51,14 @@ protected void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
 
         navigationController = new HomepageFragmentController(getSupportFragmentManager(), R.id.container);
-        navigationController.loadHomeInitially();
+        if (savedInstanceState == null) {
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java:54]: We should annotate savedInstanceState as @Nullable.

File: app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java [hunk 4]
```diff
@@ -29,6 +29,8 @@
 public class HomeActivity extends StandardActivity implements
         BottomNavigationView.Listener, CreateListDialog.Listener, PaymentManager.Listener {
 
+    private static final String PREVIOUS_SELECTED_ID = "previousSelectedId";
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java:32]: This is more descriptive IMHO.
```suggestion
    private static final String PREVIOUSLY_SELECTED_PAGE_ID = "previouslySelectedPageId";
```

File: app/src/main/java/com/randomappsinc/studentpicker/home/HomepageFragmentController.java [hunk 5]
```diff
@@ -53,10 +52,15 @@ void onNavItemSelected(@IdRes int viewId) {
...
     /** Called by the app upon start up to load the homepage */
     void loadHomeInitially() {
+        this.homepageFragment = HomepageFragment.getInstance();
         currentViewId = R.id.home;
         addFragment(homepageFragment);
     }
 
+     int getCurrentViewId() {
+        return currentViewId;
+    }
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/home/HomepageFragmentController.java:62]:
```suggestion
    int getCurrentViewId() {
        return currentViewId;
    }
```

**Review by Gear61 — APPROVED** [review]
Good first step! I just have some small comments you can dress in the next pull request.

**Pull request merged**